### PR TITLE
[SYCL] Update invoke_simd smoke test, add test on argument conversion.

### PR DESCRIPTION
**Pull request by kbobrovs**
`invoke_simd_smoke.cpp` is updated due to compiler changes.
`[[intel::device_indirectly_callable]]` attribute is now required for
user functions which are invoke_simd call targets.

Complementary change for https://github.com/intel/llvm/pull/6544.

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/InvokeSimd/invoke_simd_conv.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  }
+  int err_cnt = 0;
+
+  for (unsigned i = 0; i < Size; ++i) {
+    unsigned group_id = i / GroupSize;
+    uint32_t sg_id = (i - (group_id * GroupSize)) / VL;
+    SimdElemT test = A[i];
+    SimdElemT gold = calc((SimdElemT)sg_id);
+    if (test != gold) {
+      if (++err_cnt < 10) {
```
Comment (steffenlarsen) [inline SYCL/InvokeSimd/invoke_simd_conv.cpp]: Nit but these might as well be merged into one if-statement to save an indention level.

**Review by steffenlarsen — DISMISSED** [review]
LGTM! Only a small comment.

File: SYCL/InvokeSimd/invoke_simd_conv.cpp (hunk 1 above)
Comment (kbobrovs) [reply SYCL/InvokeSimd/invoke_simd_conv.cpp]: Thanks, updated.

**Review by steffenlarsen — APPROVED** [review]

**Comment by kbobrovs** [comment]
Failures are unrelated:

Jenkins/llvm-test-suite:
[2022-08-16T00:48:58.709Z] Unexpectedly Passed Tests (2):
[2022-08-16T00:48:58.709Z]   SYCL :: SubGroup/reduce_spirv13.cpp
[2022-08-16T00:48:58.709Z]   SYCL :: SubGroup/reduce_spirv13_fp64.cpp

Failures are expected and fixed by the compiler patch:
Jenkins/pre-ci-linux:
[2022-08-15T20:25:30.587Z] Failed Tests (1):
[2022-08-15T20:25:30.587Z]   SYCL :: InvokeSimd/invoke_simd_conv.cpp

**Pull request merged**